apps/dashboard/src/hooks/use-update-organization-settings.ts

```diff
@@ -16,10 +16,33 @@ export function useUpdateOrganizationSettings() {
     mutationFn: async (data) => {
       return updateOrganizationSettings({ data, environment: currentEnvironment! });
     },
-    onSuccess: () => {
-      queryClient.invalidateQueries({ queryKey: [QueryKeys.organizationSettings, currentEnvironment?._id] });
+    onMutate: async (newSettings) => {
+      const queryKey = [QueryKeys.organizationSettings, currentEnvironment?._id];
+
+      // Optimistically update the cache
+      const previousData = queryClient.getQueryData<{ data: GetOrganizationSettingsDto }>(queryKey);
+
+      if (previousData) {
+        queryClient.setQueryData(queryKey, {
+          ...previousData,
+          data: {
+            ...previousData.data,
+            ...newSettings,
+          },
+        });
+      }
+    },
+    onSuccess: async (response) => {
+      const queryKey = [QueryKeys.organizationSettings, currentEnvironment?._id];
+
+      // Update with the actual server response
+      await queryClient.setQueryData(queryKey, response);
     },
     onError: (error) => {
+      // Just invalidate on error to refetch the correct data
+      const queryKey = [QueryKeys.organizationSettings, currentEnvironment?._id];
+      queryClient.invalidateQueries({ queryKey });
+
```

Comment:
Optimistic update to prevent visual glitch when switching the toggle in settings